### PR TITLE
LPS-158069 Fix objects in case of deleting default value from object field picklist

### PR DESCRIPTION
**Pull request by M-IgorFranca**

### CODE REVIEW

**Comment by MarinhoFeliphe** [comment]
Hi @murilostodolni and @M-IgorFranca 

could you please remove the changes related to rebase:

https://github.com/liferay-objects/liferay-portal/pull/924/files#diff-a50d68735e9840b6a835fed264e1b77700dcf952afe1eb61018ab8462a691a16R496

Thank you!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hi @murilostodolni and @M-IgorFranca

could you please remove the changes related to rebase:

https://github.com/liferay-objects/liferay-portal/pull/924/files#diff-a50d68735e9840b6a835fed264e1b77700dcf952afe1eb61018ab8462a691a16R496

Thank you!

**Comment by MarinhoFeliphe** [comment]
Feel free to contact me If you need help to resolve this conflict.

**Comment by M-IgorFranca** [comment]
Hi @MarinhoFelipe , we managed to solve this, thanks!

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Comment by M-IgorFranca** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx [hunk 1]
```diff
@@ -107,9 +108,31 @@ export default function ObjectFieldFormBase({
...
 
+	useEffect(() => {
+		const {objectFieldSettings} = values;
+
+		if (values.businessType === 'Picklist' && objectFieldSettings) {
+			const [objectFieldPicklist] = objectFieldSettings;
+			const {value} = objectFieldPicklist;
+			const {objectStates} = value as ObjectFieldPicklistSetting;
+			const defaultPicklistValue = objectStates.find(
+				(objectState) => objectState.key === values.defaultValue
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx:124]: @M-IgorFranca, you can extract the key directly like:

```ts
const defaultPicklistValue = objectStates.find(
    ({key}) => key === defaultValue
)
```

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @M-IgorFranca, Great job on this PR, I have some suggestions, feel free to come to me and ask if you have any questions. :smile:

**Comment by M-IgorFranca** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx [hunk 2]
```diff
@@ -107,9 +108,31 @@ export default function ObjectFieldFormBase({
 		return businessTypeMap;
 	}, [objectFieldTypes]);
 
+	const [defaultValue, setDefaltValue] = useState<ObjectState>();
+	const [defaultValueQuery, setDefaultValueQuery] = useState<string>('');
```
[the remark this answers is not in the document]
Comment (M-IgorFranca) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx]: It makes sense.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx [hunk 3]
```diff
@@ -107,9 +108,31 @@ export default function ObjectFieldFormBase({
...
 	const [pickLists, setPickLists] = useState<PickList[]>([]);
 	const [pickListItems, setPickListItems] = useState<PickListItem[]>([]);
 
+	useEffect(() => {
+		const {objectFieldSettings} = values;
```
[the remark this answers is not in the document]
Comment (M-IgorFranca) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx]: Done :)

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx [hunk 4]
```diff
@@ -107,9 +108,31 @@ export default function ObjectFieldFormBase({
...
+
+		if (values.businessType === 'Picklist' && objectFieldSettings) {
+			const [objectFieldPicklist] = objectFieldSettings;
+			const {value} = objectFieldPicklist;
```
[the remark this answers is not in the document]
Comment (M-IgorFranca) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx]: Done 👍🏾

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx (hunk 1 above)
Comment (M-IgorFranca) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx]: Done 👍🏾

**Review by danvb112 — APPROVED** [review]
LGTM!

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx [hunk 5]
```diff
@@ -107,9 +108,32 @@ export default function ObjectFieldFormBase({
 		return businessTypeMap;
 	}, [objectFieldTypes]);
 
+	const [picklistDefaultValue, setPicklistDefaultValue] = useState<
+		ObjectState
+	>();
+	const [defaultValueQuery, setDefaultValueQuery] = useState<string>('');
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx:114]: Hey @M-IgorFranca, Could you rename this state name to `picklistDefaultValueQuery` and `setPicklistDefaultValueQuery` too?

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @M-IgorFranca, sorry a have only one more suggestion

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx (hunk 5 above)
Comment (M-IgorFranca) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx]: Its make sense, sorry!

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Review by MarinhoFeliphe — APPROVED** [review]

**Comment by MarinhoFeliphe** [comment]
Resent [here](https://github.com/brianchandotcom/liferay-portal/pull/122241).

**Pull request closed without merging**